packages/malloy/src/model/malloy_query.ts

```diff
@@ -4441,6 +4471,24 @@ class QueryStruct extends QueryNode {
             false
           );
           return dtStageWriter.addPDT(name, this.dialect);
+        } else if (sourceTag.has('materialize')) {
```

Comment:
We probably want to make a setting that is passed into the compiler of whether the compiler should produce the normal SQL or the "depends-on-materialized-stuff" SQL.